CommandLine/libGMX/gmx.cpp

```diff
@@ -0,0 +1,382 @@
+/** Copyright (C) 2018 Greg Williamson, Robert B. Colton
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include <pugixml.hpp>
+
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <vector>
+#include <algorithm>
+
+#include "gmx.h"
+
+namespace gmx {
+
+inline std::string string_replace_all(std::string str, std::string substr, std::string nstr)
```

Comment:
Put `namespace {}` around this static stuff.

Reply:
Are you talking about making them translation unit local so they do not clash with the copy that's also in CompilerSource at linktime? How is not ok with them already being in the GMX namespace? You are afraid of somebody adding `using namespace gmx;` somewhere in CompilerSource (which could very well happen as we move proto2es and deprecated backend/)?